Firestore/core/src/firebase/firestore/local/leveldb_key.cc

```diff
@@ -322,6 +342,26 @@ class Reader {
     return ReadInt32();
   }
 
+  /**
+   * Reads a signed number from the key.
+   *
+   * If the read is unsuccessful, returns 0 and fails the Reader.
+   *
+   * Otherwise, returns the number and advances the Reader to the next unread
+   * byte.
+   */
+  int32_t ReadInt64() {
+    if (ok_) {
```

Comment:
`ReadSignedNumIncreasing` already does this dance so we don't need these additional checks. This could really just be `return ReadSignedNumIncreasing();`. `ReadInt32` does additional work (a checked cast) so it can't just be a pass-through.

Alternatively, you could just call `ReadSignedNumIncreasing` directly in the caller and omit this.

Reply:
I left the method to make the callsites more obvious, but removed the additional ok_ checks.